pilot/pkg/config/kube/ingress/status.go

```diff
@@ -144,7 +156,7 @@ func (s *StatusSyncer) runningAddresses() []string {
 
 	for _, pod := range igPods {
 		// only Running pods are valid
-		if pod.Status.Phase != corev1.PodRunning {
+		if pod.Status.Phase != corev1.PodRunning || s.nodes == nil {
```

Comment:
You can see below the following piece of code:
```
		// Find node external IP
		node := s.nodes.Get(pod.Spec.NodeName, "")
		if node == nil {
			continue
		}
```
First of all, `s.nodes` is not set in multi-tenant mode, because then istiod has no permissions to perform any operation on `Node` objects. So this condition ensures that we don't reach that peace of code and istiod will not try to get nodes when it has no proper permissions.
I agree that it's a little bit confusing, but I didn't care too much about readability, because I understand all these hacks... I can extract and comment this condition to make it more clear.

Reply:
I think a comment on `why` might be helpful.

Reply:
I noticed that when `s.nodes == nil`, iterating over ingress pods does not make sense, because we will end up in `continue` in all iterations, so I returned earlier and skipped getting ingress pods.